### ISSUE TITLE
AWS installer hangs after spinning up flynn's system apps

### ISSUE THREAD

**Issue by swaits** [flynn/flynn#3017]
When running `flynn install` to AWS it hangs at the end of the installation.

* used the browser interface
* US East, 3x m4.large
* included a cluster backup:
  * 3.3GB
  * uploaded via browser

Version:

```
% flynn version
v20160627.0
```

```
ubuntu@ip-10-0-4-134:~$ flynn-host version
v20160624.1
```

Host debug dump: 

https://gist.github.com/anonymous/4742975e04a78121d345d295154bd147

Installer log:

```
Using saved key pair (flynn-dzap)
Allocating domain
Fetching image manifest
Generating start script
Creating stack
AWS::CloudFormation::Stack (flynn-1467137329)	CREATE_IN_PROGRESS (User Initiated)
AWS::EC2::InternetGateway (Gateway)	CREATE_IN_PROGRESS
AWS::EC2::VPC (VPC)	CREATE_IN_PROGRESS
AWS::EC2::InternetGateway (Gateway)	CREATE_IN_PROGRESS (Resource creation Initiated)
AWS::EC2::VPC (VPC)	CREATE_IN_PROGRESS (Resource creation Initiated)
AWS::Route53::HostedZone (DNSZone)	CREATE_IN_PROGRESS
AWS::Route53::HostedZone (DNSZone)	CREATE_IN_PROGRESS (Resource creation Initiated)
AWS::EC2::InternetGateway (Gateway)	CREATE_COMPLETE
AWS::EC2::VPC (VPC)	CREATE_COMPLETE
AWS::EC2::VPCGatewayAttachment (GatewayAttachment)	CREATE_IN_PROGRESS
AWS::EC2::RouteTable (GatewayRouteTable)	CREATE_IN_PROGRESS
AWS::EC2::VPCGatewayAttachment (GatewayAttachment)	CREATE_IN_PROGRESS (Resource creation Initiated)
AWS::EC2::RouteTable (GatewayRouteTable)	CREATE_IN_PROGRESS (Resource creation Initiated)
AWS::EC2::SecurityGroup (PublicSecurityGroup)	CREATE_IN_PROGRESS
AWS::EC2::Subnet (Subnet)	CREATE_IN_PROGRESS
AWS::EC2::Subnet (Subnet)	CREATE_IN_PROGRESS (Resource creation Initiated)
AWS::EC2::RouteTable (GatewayRouteTable)	CREATE_COMPLETE
AWS::Route53::HostedZone (DNSZone)	CREATE_COMPLETE
AWS::EC2::VPCGatewayAttachment (GatewayAttachment)	CREATE_COMPLETE
AWS::EC2::SecurityGroup (PublicSecurityGroup)	CREATE_IN_PROGRESS (Resource creation Initiated)
AWS::EC2::Subnet (Subnet)	CREATE_COMPLETE
AWS::EC2::Route (GatewayRoute)	CREATE_IN_PROGRESS
AWS::EC2::Route (GatewayRoute)	CREATE_IN_PROGRESS (Resource creation Initiated)
AWS::EC2::SubnetRouteTableAssociation (SubnetRoute)	CREATE_IN_PROGRESS
AWS::EC2::SubnetRouteTableAssociation (SubnetRoute)	CREATE_IN_PROGRESS (Resource creation Initiated)
AWS::EC2::SecurityGroup (PublicSecurityGroup)	CREATE_COMPLETE
AWS::EC2::Instance (Instance0)	CREATE_IN_PROGRESS
AWS::EC2::Instance (Instance1)	CREATE_IN_PROGRESS
AWS::EC2::Instance (Instance2)	CREATE_IN_PROGRESS
AWS::EC2::Instance (Instance0)	CREATE_IN_PROGRESS (Resource creation Initiated)
AWS::EC2::Instance (Instance2)	CREATE_IN_PROGRESS (Resource creation Initiated)
AWS::EC2::Instance (Instance1)	CREATE_IN_PROGRESS (Resource creation Initiated)
AWS::EC2::Route (GatewayRoute)	CREATE_COMPLETE
AWS::EC2::SubnetRouteTableAssociation (SubnetRoute)	CREATE_COMPLETE
AWS::EC2::Instance (Instance0)	CREATE_COMPLETE
AWS::EC2::Instance (Instance1)	CREATE_COMPLETE
AWS::EC2::Instance (Instance2)	CREATE_COMPLETE
AWS::Route53::HealthCheck (Instance0HealthCheck)	CREATE_IN_PROGRESS
AWS::Route53::HealthCheck (Instance0HealthCheck)	CREATE_IN_PROGRESS (Resource creation Initiated)
AWS::Route53::HealthCheck (Instance0HealthCheck)	CREATE_COMPLETE
AWS::Route53::HealthCheck (Instance2HealthCheck)	CREATE_IN_PROGRESS
AWS::Route53::HealthCheck (Instance1HealthCheck)	CREATE_IN_PROGRESS
AWS::Route53::HealthCheck (Instance2HealthCheck)	CREATE_IN_PROGRESS (Resource creation Initiated)
AWS::Route53::HealthCheck (Instance1HealthCheck)	CREATE_IN_PROGRESS (Resource creation Initiated)
AWS::Route53::HealthCheck (Instance2HealthCheck)	CREATE_COMPLETE
AWS::Route53::HealthCheck (Instance1HealthCheck)	CREATE_COMPLETE
AWS::Route53::RecordSetGroup (DNSRecords)	CREATE_IN_PROGRESS
AWS::Route53::RecordSetGroup (DNSRecords)	CREATE_IN_PROGRESS (Resource creation Initiated)
AWS::Route53::RecordSetGroup (DNSRecords)	CREATE_COMPLETE
AWS::CloudFormation::Stack (flynn-1467137329)	CREATE_COMPLETE
Fetching stack
Configuring DNS
Uploading backup
Running bootstrap
online-hosts: start
online-hosts: done
discoverd: start
discoverd: done
flannel: start
flannel: done
wait-hosts: start
wait-hosts: done
postgres: start
postgres: done
postgres-wait: start
postgres-wait: done
mariadb: start
mariadb: done
mariadb-wait: start
mariadb-wait: done
restore: start
restore: done
restore: start
restore: done
controller: start
controller: done
wait-controller: start
wait-controller: done
blobstore: start
blobstore: done
blobstore-wait: start
blobstore-wait: done
controller-scheduler: start
controller-scheduler: done
status: start
status: done
cluster-monitor: start
cluster-monitor: done
```

**Comment by titanous** [comment]
@jvatic Any ideas about what happened here? Given the output, I believe bootstrap finished and exited successfully.

**Comment by jvatic** [comment]
@titanous it could be related to #2987 as restoring from backup performs a domain migration after bootstrapping.

**Comment by swaits** [comment]
@jvatic @titanous here's a log and stack dump that may help:

https://gist.github.com/swaits/fdeeb76f6014777b6b18cf017dc2496c

**Comment by swaits** [comment]
Fixed, for me, in `v20160701.0`. Thanks!
